src/controllers/routers/cases.js

```diff
@@ -74,14 +73,194 @@ function sanitizedCase(json) {
+ * @return {Object} query - Query params with default values
+ */
+function setDefault(json) {
+  const query = json;
+  // if with_paging is not available or false, set page and per_page to null
+  if (!json.with_paging || json.with_paging !== 'true') {
+    query.page = null;
+    query.per_page = null;
+  }
+  // if with_paging is true
+  else {
+    // if page is defined in query param, otherwise default to 1
+    if (json.page !== undefined || json.page !== null) {
+      query.page = parseInt(json.page, 10);
+    } else {
+      query.page = 1;
+    }
+    // if per_page is defined in query param, otherwise default to 10
+    if (json.per_page !== undefined || json.per_page !== null) {
+      query.per_page = parseInt(json.per_page, 10);
+    } else {
+      query.per_page = 10;
+    }
+  }
+  // if status is not present, set default to 'ALL'
+  if (!json.status) {
+    query.status = 'ALL';
+  }
+  // if sort is not present, set default to arrange id in asc
+  if (!json.sort) {
+    query.sort_field = 'id';
+    query.sort_order = 'asc';
+  }
+  return query;
+}
+
 /**
  * Retrieve all cases
  * @param {Request} req
  * @param {Response} res
  */
-const getAll = async (req, res) => {
-  const cases = await Case.query().select();
-  return res.status(200).json({ cases });
+const getAll = async (req, res, next) => {
+  const parsedQueries = sanitizedQuery(req.query);
+
+  // to set the default values for with_paging, page, per_page and status
+  setDefault(parsedQueries);
+
+  const currentPage = parsedQueries.page;
+  const limit = parsedQueries.per_page;
+  // if offset is undefined, set it to 0
+  const offset = limit * currentPage - limit ? limit * currentPage - limit : 0;
+
+  // to retrieve all cases to find the total number of cases for the response object
+  const allCases = await Case.query();
+  try {
+    const results = await Case.query()
+      .withGraphJoined(parsedQueries.include_entities)
+      .where((builder) => {
+        if (parsedQueries.beneficiary_name) {
+          builder.where(
+            'beneficiary.name',
+            'like',
+            `%${parsedQueries.beneficiary_name}%`
+          );
+        }
+        if (parsedQueries.referee_name) {
+          builder.where(
+            'referee.name',
+            'like',
+            `%${parsedQueries.referee_name}%`
+          );
+        }
+        if (parsedQueries.referee_org) {
+          builder.where(
+            'referee.organisation',
+            'like',
+            `%${parsedQueries.referee_org}%`
+          );
+        }
+        if (parsedQueries.status !== 'ALL') {
+          builder.where('caseStatus', parsedQueries.status);
+        }
+        if (parsedQueries.case_number) {
+          builder.where('caseNumber', parsedQueries.case_number);
+        }
+        if (parsedQueries.applied_on) {
+          builder.where('appliedOn', parsedQueries.applied_on);
+        }
+      })
+      .orderBy(parsedQueries.sort_field, parsedQueries.sort_order)
```

Comment:
this does not seem to support the sort fields as specified in Figma. For example, how do I sort by Beneficiary Name?

Reply:
update to include sort for all the available fields in figma (id, beneficiaryName, appliedOn, poc, referenceStatus, refereeName, refereeOrganisation, updatedBy)
Current updatedBy is sorted by staff id